configuration/components/loki.libsonnet

```diff
@@ -285,34 +277,19 @@ function(params) {
           // Defaults: https://github.com/grafana/loki/blob/main/production/ksonnet/loki/config.libsonnet#L181-L184
           frontend+: {
             // Mixins don't support prefixes in resources names so we have to overwrite
-            // local schedulerService = newService('query_scheduler'),
-            // scheduler_address: '%s.%s.svc.cluster.local:9095' % [schedulerService.metadata.name, loki.config.namespace],
-            // TODO(JoaoBraveCoding): replace when working on services
-            scheduler_address: '%s.%s.svc.cluster.local:9095' % [
-              loki.config.name + '-query-scheduler-grpc',
-              loki.config.namespace,
-            ],
+            local schedulerService = newService('query_scheduler'),
+            scheduler_address: '%s.%s.svc.cluster.local:9095' % [schedulerService.metadata.name, loki.config.namespace],
             // Mixins don't support tail_proxy_url, when support is added we will
             // still need support for prefixing resource names before removing this
-            // local querierService = newService('querier'),
-            // tail_proxy_url: '%s.%s.svc.cluster.local:3100' % [querierService.metadata.name, loki.config.namespace],
-            // TODO(JoaoBraveCoding): replace when working on services
-            tail_proxy_url: '%s.%s.svc.cluster.local:3100' % [
-              loki.config.name + '-querier-http',
-              loki.config.namespace,
-            ],
+            local querierService = newService('querier'),
+            tail_proxy_url: '%s.%s.svc.cluster.local:3100' % [querierService.metadata.name, loki.config.namespace],
           },
           // Docs: https://grafana.com/docs/loki/latest/configuration/#frontend_worker
           // Defaults: https://github.com/grafana/loki/blob/main/production/ksonnet/loki/config.libsonnet#L185-L190
           frontend_worker+: {
             // Mixins dont support prefixes in resources names so we have to overwrite
-            // local schedulerService = newService('query_scheduler'),
-            // scheduler_address: '%s.%s.svc.cluster.local:9095' % [schedulerService.metadata.name, loki.config.namespace],
-            // TODO(JoaoBraveCoding): replace when working on services
-            scheduler_address: '%s.%s.svc.cluster.local:9095' % [
-              loki.config.name + '-query-scheduler-grpc',
-              loki.config.namespace,
-            ],
+            local schedulerService = newService('query_scheduler'),
+            scheduler_address: '%s.%s.svc.cluster.local:9095' % [schedulerService.metadata.name, loki.config.namespace],
```

Comment:
Same here, shouldn't that be the headless service for the scheduler?

Reply:
According to the docs the `frontend_worker` block to discover scheduler pods is not part of memberlist, it simplified by using the DNS SRV records and leave anything else to GRPC: https://grafana.com/docs/loki/v2.8.x/configuration/#frontend_worker

```yaml
# Hostname (and port) of scheduler that querier will periodically resolve,
# connect to and receive queries from. Only one of -querier.frontend-address or
# -querier.scheduler-address can be set. If neither is set, queries are only
# received via HTTP endpoint.
# CLI flag: -querier.scheduler-address
[scheduler_address: <string> | default = ""]
```